src/v/redpanda/request_auth.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2022 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/fwd.h"
+#include "config/property.h"
+
+#include <seastar/http/request.hh>
+
+#include <security/credential_store.h>
+
+/**
+ * Helper for HTTP request handlers that would like to enforce
+ * authentication and authorization rules.
+ *
+ * Authentication is done on construction: either basic auth or mtls is
+ * accepted.  If neither succeeds, an http response exception will be thrown.
+ *
+ * Authorization is done using one of the require_* methods: e.g.
+ * require_superuser.
+ */
+class request_auth_result {
+public:
+    /**
+     * Authenticated user.  They have passed authentication so we know
+     * their identity, and whether that identity is a superuser.
+     *
+     * @param username
+     * @param is_superuser
+     */
+    request_auth_result(security::credential_user username, bool is_superuser)
+      : _username(username)
+      , _authenticated(true)
+      , _superuser(is_superuser){};
+
+    /**
+     * Anonymous user.  They may still be considered authenticated/superuser
+     * if the global require_auth property is set to false (i.e. anonymous
+     * users have all powers)
+     */
+    request_auth_result(bool is_authenticated, bool is_superuser)
+      : _authenticated(is_authenticated)
+      , _superuser(is_superuser){};
+
+    ~request_auth_result() noexcept(false);
+
+    /**
+     * Raise 403 if not a superuser
+     */
+    void require_superuser();
+
+    /**
+     * Raise 403 if not authenticated
+     */
+    void require_authenticated();
+
+    /**
+     * Do nothing.  Hook for logging access on un-authenticated API endpoints.
+     */
+    void pass();
+
+    ss::sstring const& get_username() { return _username; }
+
+private:
+    security::credential_user _username;
+    bool _authenticated{false};
+    bool _superuser{false};
+    bool _checked{false};
```

Comment:
I think the API would be clearer and safer to use if these were `ss::bool_class`.